pkg/controller/plan/kubevirt.go

```diff
@@ -969,6 +937,130 @@ func (r *KubeVirt) DeleteHookJobs(vm *plan.VMStatus) (err error) {
 	return
 }
 
+// Delete any populator resources that belong to a VM migration.
+func (r *KubeVirt) DeletePopulatorResources(vm *plan.VMStatus) (err error) {
+	if r.isOpenstack(vm) {
+		err = r.deleteOpenstackVolumePopulatorCrList(vm)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+	} else if r.useOvirtPopulator(vm) {
+		err = r.deleteOvirtPopulatorCrList(vm)
+		if err != nil {
+			return liberr.Wrap(err)
+		}
+	}
+	return
+}
+
+// Get the OpenstackVolumePopulator CustomResource based on the image name.
+func (r *KubeVirt) getOpenstackPopulatorCr(name string) (populatorCr v1beta1.OpenstackVolumePopulator, err error) {
+	populatorCr = v1beta1.OpenstackVolumePopulator{}
+	err = r.Client.Get(context.TODO(), client.ObjectKey{Namespace: r.Plan.Spec.TargetNamespace, Name: name}, &populatorCr)
+	if err != nil {
+		return
+	}
+	return
+}
+
+// Get the OpenstackVolumePopulator CustomResource List.
+func (r *KubeVirt) getOpenstackPopulatorCrList() (populatorCrList v1beta1.OpenstackVolumePopulatorList, err error) {
+	populatorCrList = v1beta1.OpenstackVolumePopulatorList{}
+	err = r.Client.List(
+		context.TODO(),
+		&populatorCrList,
+		&client.ListOptions{
+			Namespace:     r.Plan.Spec.TargetNamespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{kMigration: string(r.Plan.Status.Migration.ActiveSnapshot().Migration.UID)}),
```

Comment:
why not `r.Migration.UID`?

Reply:
because `r.Migration.UID` isn't populated when the plan is not executed. meaning, if we try to clear it afterwards and query `r.Migration.UID` is nil.